### PR TITLE
SED-1998 [OS] Migrate edit scheduler task dialog

### PR DESCRIPTION
**Pull request by dvladir**

### CODE REVIEW

File: projects/step-core/src/lib/components/select-plan/select-plan.component.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
-<step-form-field>
-  <step-label show-required-marker>Plan</step-label>
-  <input type="text" [formControl]="planNameControl" [tabIndex]="-1" />
+<step-form-field [class.ng-invalid]="_ngControl.invalid" [class.ng-touched]="_ngControl.touched">
+  <step-label [attr.show-required-marker]="showRequiredMarker || null">Plan</step-label>
+  <input type="text" readonly [disabled]="isDisabled" [ngModel]="planValue?.attributes?.['name']" [tabIndex]="-1" />
   <step-suffix class="button-suffix">
-    <button type="button" mat-stroked-button (click)="selectPlan()" [disabled]="disabled">Select</button>
+    <button type="button" mat-stroked-button (click)="selectPlan()" [disabled]="isDisabled">Select</button>
```
Comment (bojidaryovchev) [inline projects/step-core/src/lib/components/select-plan/select-plan.component.html:5]: also for the coding style guide - ```disabled``` vs ```isDisabled```

File: projects/step-core/src/lib/components/edit-scheduler-task-dialog/edit-scheduler-task-dialog.component.html [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      </step-form-field>
+    </form>
+    <mat-expansion-panel [(expanded)]="showParameters">
+      <mat-expansion-panel-header>
+        <mat-panel-title>Advanced</mat-panel-title>
+      </mat-expansion-panel-header>
+      <section class="advanced-body">
+        <step-form-field>
+          <step-label>Parameters</step-label>
+          <textarea
```
Comment (bojidaryovchev) [inline projects/step-core/src/lib/components/edit-scheduler-task-dialog/edit-scheduler-task-dialog.component.html:46]: I think this should use a reactive form and have a JSON validator which would then display an error below if the JSON is invalid (I've done that as part of the function configuration dialog migration)

Comment (dvladir) [reply projects/step-core/src/lib/components/edit-scheduler-task-dialog/edit-scheduler-task-dialog.component.html:46]: @bojidaryovchev This editor uses `step-custom-froms` component, that is downgraded, is used in legacy UI and doesn't support reactive forms. That's why I decided to use template driven form for this dialog.

File: projects/step-core/src/lib/components/edit-scheduler-task-dialog/edit-scheduler-task-dialog.component.html [hunk 3]
```diff
@@ -0,0 +1,61 @@
+<step-modal-window [title]="isNew ? 'New task' : 'Edit task'">
+  <div step-modal-window-body>
+    <step-alert *ngIf="error" type="danger">
```
Comment (bojidaryovchev) [inline projects/step-core/src/lib/components/edit-scheduler-task-dialog/edit-scheduler-task-dialog.component.html:3]: another thing we should consider - say we have an ```enum Foo { BAR = 'bar' }``` and component ```app-something``` which has an ```@Input() foo!: Foo;```

so should we do it like ```<app-something foo="bar"></app-something``` or ```<app-something [foo]="Foo.BAR"></app-something>``` (having ```protected readonly Foo = Foo``` in the ```.ts```)

File: projects/step-core/src/lib/components/edit-scheduler-task-dialog/edit-scheduler-task-dialog.component.html (hunk 2 above)
Comment (bojidaryovchev) [reply projects/step-core/src/lib/components/edit-scheduler-task-dialog/edit-scheduler-task-dialog.component.html:46]: @dvladir I also used this component in the function configuration dialog

**Review by bojidaryovchev — APPROVED** [review]

File: projects/step-core/src/lib/components/edit-scheduler-task-dialog/edit-scheduler-task-dialog.component.html (hunk 2 above)
Comment (bojidaryovchev) [reply projects/step-core/src/lib/components/edit-scheduler-task-dialog/edit-scheduler-task-dialog.component.html:46]: image

Comment (bojidaryovchev) [reply projects/step-core/src/lib/components/edit-scheduler-task-dialog/edit-scheduler-task-dialog.component.html:46]: image

Comment (bojidaryovchev) [reply projects/step-core/src/lib/components/edit-scheduler-task-dialog/edit-scheduler-task-dialog.component.html:46]: image

Comment (bojidaryovchev) [reply projects/step-core/src/lib/components/edit-scheduler-task-dialog/edit-scheduler-task-dialog.component.html:46]: image

**Review by bojidaryovchev — APPROVED** [review]

**Pull request merged**